vision/emergent_object/emergent_object_processing.py

```diff
@@ -0,0 +1,161 @@
+
+    subject_humanoid: BoundingBox
+    for subject_humanoid in humanoids:
+        # Get the distance to the closest standard object
+        min_odlc_distance: float = float("inf")
+        odlc: Location
+        for odlc in odlcs.values():
+            odlc_distance: float = geopy.distance.geodesic(
+                (odlc["latitude"], odlc["longitude"]),
+                (subject_humanoid.attributes["latitude"], subject_humanoid.attributes["longitude"]),
+            ).feet
+
+            min_odlc_distance = min(odlc_distance, min_odlc_distance)
+
+        # Get the distance to the closest emergent object that's not in the same image
+        min_humanoid_distance: float = float("inf")
+        humanoid: BoundingBox
+        for humanoid in humanoids:
+            # Don't check candidates that are in the same image
+            if subject_humanoid.attributes["image_path"] != humanoid.attributes["image_path"]:
+                humanoid_distance: float = geopy.distance.geodesic(
+                    (humanoid.attributes["latitude"], humanoid.attributes["longitude"]),
+                    (
+                        subject_humanoid.attributes["latitude"],
+                        subject_humanoid.attributes["longitude"],
+                    ),
+                ).feet
+
+                min_humanoid_distance = min(humanoid_distance, min_humanoid_distance)
+
+        # invert the humanoid distance to prioritize closest humanoids
+        min_humanoid_distance *= -1
+
+        evaluation: tuple[float, float, float, float] = (
+            min_odlc_distance,
+            min_humanoid_distance,
+            subject_humanoid.attributes["bounding_box_area"],
+            subject_humanoid.attributes["confidence"],
+        )
+
+        evaluations_list.append(evaluation)
+
+    evaluation_array: NDArray[Shape["*, 4"], Float64] = np.array(evaluations_list)
+
+    # Get the minimums of each category
+    evaluation_mins: NDArray[Shape["4"], Float64] = np.amin(evaluation_array, axis=0)
+
+    # Get the maximums of each category
+    evaluation_maxes: NDArray[Shape["4"], Float64] = np.amax(evaluation_array, axis=0)
+
+    # An error here could mean that all candidates were detected in the same image - This
+    #   is okay to ignore
+    with np.errstate(invalid="ignore"):
+        ranges: NDArray[Shape["4"], Float64] = evaluation_maxes - evaluation_mins
+
+    # Suppress warnings when getting 0/0 errors - This is okay because that means that
+    #   all candidates scored the same in that category
+    with np.errstate(invalid="ignore"):
+        normalized_evaluations: NDArray[Shape["*, 4"], Float64] = (
+            evaluation_array - evaluation_mins
+        ) / ranges
+
+    weighted_evaluations: NDArray[Shape["*, 4"], Float64] = (
+        normalized_evaluations * category_weights
+    )
+
+    # Use nansum to ignore categories with divide by zero errors
+    scores: NDArray[Shape["*"], Float64] = np.nansum(weighted_evaluations, axis=1)
```

Comment:
`nansum` is used to treat `NaN` values as 0. If there are `NaN` values in the array, wouldn't you also need to use `nanprod` for the multiplication? Also, using `nanprod` would mean there are no `NaN` values in the array, so you could do a regular `sum`.

Another alternative would be to do your division above with `np.divide`, adding a where clause to remove `NaN` values: https://numpy.org/doc/stable/reference/generated/numpy.divide.html . This would also avoid having the error state at all.